typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -266,21 +271,58 @@ export abstract class HyperlaneDeployer<
   ): Promise<ReturnType<F['deploy']>> {
     const cachedContract = this.readCache(chain, factory, contractName);
     if (cachedContract) {
+      const provider = this.multiProvider.getProvider(chain);
```

Comment:
Should we check this in? Might be nice to put this in a separate function and then comment calling it out if we need this in the future? Also, it would just duplicate them now otherwise no?